src/server.c

```diff
@@ -158,7 +172,7 @@ void handle_http_request(int fd, struct cache *cache)
 {
     const int request_buffer_size = 65536; // 64K
     char request[request_buffer_size];
-    char method[512]; // GET or POST
+    char method[5]; // GET or POST
```

Comment:
I hadn't noticed this yesterday, but good change. Technically speaking if you REALLY wanted to stretch you could incorporate DELETE, and then your buffer would be a little bigger, but this is a good number for GET and POST.